lib/git_reflow/git_server/git_hub/pull_request.rb

```diff
@@ -78,6 +78,45 @@ def approved?
           end
         end
 
+        def merge_feature_branch(feature_branch_name, options = {})
+          options[:destination_branch] ||= 'master'
+
+          message = "#{options[:message]}"
+
+          if "#{options[:pull_request_number]}".length > 0
+            message << "\nMerges ##{options[:pull_request_number]}\n"
+          end
+
+          if lgtm_authors = Array(options[:lgtm_authors]) and lgtm_authors.any?
+            message << "\nLGTM given by: @#{lgtm_authors.join(', @')}\n"
+          end
+
+          data = {
+            "commit_title" => options[:title],
+            "commit_message" => options[:message],
+            "sha" => options[:head].sha,
+            "squash" => true
+          }
+          
+          res = GitReflow::GitServer::GitHub.connection.pull_requests.merge(
+            "#{GitReflow::Config.get('github.owner')}", 
+            "#{GitReflow::Config.get('github.repo')}", 
+            "#{options[:pull_request_number]}", 
+            data
+          )
+
+          if res.code == "405"
```

Comment:
The `github_api` gem wraps responses in a [Github::ResponseWrapper](https://github.com/piotrmurach/github/blob/master/lib/github_api/response_wrapper.rb).  We should only have to check for `merge_response.success?` (meaning `200`).  If it's not a success, we should raise a custom error with the message.  Honeybadger has a great article on [doing this in a simple way](http://blog.honeybadger.io/ruby-custom-exceptions/).  Let's throw it in a file called `lib/git_reflow/git_server/pull_request/merge_error.rb` and name the class `GitReflow::GitServer::PullRequest::MergeError` and include it in this file.  Then we can just have an unsuccessful `else` that raises our new custom error.  Something along the lines of:

```ruby
if merge_response.success?
  append_to_squashed_commit_message(message) if message.length > 0
else
  if merge_response.has_key?(:message)
    raise MergeError.new(merge_response.message)
  else
    raise MergeError.new(merge_response.to_s)
  end
end
```